public/app/features/dashboard-scene/serialization/layoutSerializers/DefaultGridLayoutSerializer.ts

```diff
@@ -142,8 +142,9 @@ function gridItemToGridLayoutItemKind(gridItem: DashboardGridItem, yOverride?: n
   width = gridItem_.state.width ?? 0;
   const repeatVar = gridItem_.state.variableName;
 
-  // FIXME: which name should we use for the element reference, key or something else ?
-  const elementName = gridItem_.state.body.state.key ?? 'DefaultName';
+  // For serialization we should retrieve the original element key
+  const elementKey = getElementIdentifierForVizPanel(gridItem.state.body) ?? 'DefaultName';
```

Comment:
`getElementIdentifierForVizPanel` will either return a string or throw an error, so we don't need nullish coalescing here

```suggestion
  const elementKey = getElementIdentifierForVizPanel(gridItem.state.body);